src/Core/Resolvers/SqlResponseHelpers.cs

```diff
@@ -0,0 +1,230 @@
+                                                                        : OkResponse(RemoveExtraFieldsInResponseWithSingleItem(findOperationResponse, extraFieldsInResponse));
+                }
+            }
+
+            List<JsonElement> rootEnumerated = findOperationResponse.EnumerateArray().ToList();
+
+            // More records exist than requested, we know this by requesting 1 extra record,
+            // that extra record is removed here.
+            rootEnumerated.RemoveAt(rootEnumerated.Count - 1);
+
+            // The fields such as primary keys, fields in $orderby clause that are retrieved in addition to the
+            // fields requested in the $select clause are required for calculating the $after element which is part of nextLink.
+            // So, the extra fields are removed post the calculation of $after element.
+            string after = SqlPaginationUtil.MakeCursorFromJsonElement(
+                               element: rootEnumerated[rootEnumerated.Count - 1],
+                               orderByColumns: context.OrderByClauseOfBackingColumns,
+                               primaryKey: sqlMetadataProvider.GetSourceDefinition(context.EntityName).PrimaryKey,
+                               entityName: context.EntityName,
+                               schemaName: context.DatabaseObject.SchemaName,
+                               tableName: context.DatabaseObject.Name,
+                               sqlMetadataProvider: sqlMetadataProvider);
+
+            // nextLink is the URL needed to get the next page of records using the same query options
+            // with $after base64 encoded for opaqueness
+            string path = UriHelper.GetEncodedUrl(httpContext!.Request).Split('?')[0];
+
+            // If the base route is not empty, we need to insert it into the URI before the rest path.
+            string? baseRoute = runtimeConfig.Runtime.BaseRoute;
+            if (!string.IsNullOrWhiteSpace(baseRoute))
+            {
+                HttpRequest request = httpContext!.Request;
+
+                // Path is of the form ....restPath/pathNameForEntity. We want to insert the base route before the restPath.
+                // Finally, it will be of the form: .../baseRoute/restPath/pathNameForEntity.
+                path = UriHelper.BuildAbsolute(
+                    scheme: request.Scheme,
+                    host: request.Host,
+                    pathBase: baseRoute,
+                    path: request.Path);
+            }
+
+            JsonElement nextLink = SqlPaginationUtil.CreateNextLink(
+                                  path,
+                                  nvc: context!.ParsedQueryString,
+                                  after);
+
+            // When there are extra fields present, they are removed before returning the response.
+            if (extraFieldsInResponse.Count > 0)
```

Comment:
Why would there be extra fields in response? Won't we take care of them while generating the `OutputColumns`? Or there would only be PKs and order by columns here since they are required for nextLink?

Reply:
I see the method summary okay.